runtime/structure.c

```diff
@@ -90,3 +35,32 @@ static bool tidListContains(const Tid *start, const Tid *end, Tid tid) {
     // start == end, so there is no such member
     return false;
 }
+
+TaggedPtr structCreateFiller(FillerDescPtr fdp, Fillability *fillability) {
+    if (fdp == NULL) {
+        *fillability = FILL_NONE;
+        return NULL;
+    }
+    bool hasIdentityPtr;
+    TaggedPtr fillerValue = filler_create(fdp, &hasIdentityPtr);
+    if (fillability != NULL) {
+        *fillability = hasIdentityPtr ? FILL_EACH : FILL_COPY;
+    }
+    return fillerValue;
+}
+
+typedef struct GenericFillerDesc {
+    TaggedPtr (*create)(struct GenericFillerDesc *fillerDesc, bool *hasIdentityPtr);
+    TaggedPtr genericValue;
+} *GenericFillerDescPtr;
+
+TaggedPtr _bal_generic_filler_create(GenericFillerDescPtr fillerDesc, bool *hasIdentityPtr) {
+    *hasIdentityPtr = false;
+    return fillerDesc->genericValue;
+}
+
+const struct GenericFillerDesc _bal_nil_filler_desc = { &_bal_generic_filler_create, NIL };
```

Comment:
Since _bal_generic_filler_create is public, it would be better to define it and GenericFillerDesc in balrt.h and then define put the type specific _bal_X_filler_desc in the corresponding .bal file.

Reply:
fixed (move the definition of `GenericFillerDesc` and declaration of `_bal_generic_filler_create` to balrt.h (definition of the create function is in structure.c). Only `_bal_nif_filler_desc` is left in `structure.c`